bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/servlets/FormMetaDataDataSourceServlet.java

```diff
@@ -155,22 +156,26 @@ private List<Resource> getDataSourceResources(HttpServletRequest request, Resour
             Iterator<FormsManager.ComponentDescription> metaDataList = null;
             switch (type) {
                 case FORMATTERS:
+                case LANG:
                     ContentPolicy policy = ComponentUtils.getPolicy((String) request.getAttribute(Value.CONTENTPATH_ATTRIBUTE),
                         resourceResolver);
                     resources.add(getResourceForDropdownDisplay(resourceResolver, "Select", ""));
                     if (policy != null) {
                         ValueMap props = policy.getProperties();
                         if (props != null) {
                             for (Map.Entry<String, Object> entry : props.entrySet()) {
-                                if (entry.getKey().startsWith(ALLOWED_FORMAT)) {
+                                if ((type.equals(FormMetaDataType.FORMATTERS) && entry.getKey().startsWith(ALLOWED_FORMAT)) ||
+                                    (type.equals(FormMetaDataType.LANG) && entry.getKey().startsWith(FormMetaDataType.LANG.getValue()))) {
```

Comment:
I would suggest making private functions and using that here for readability, something like below
```
if (isFormattersPolicy(type, entry) || isLangPolicy(type, entry))
```